### DISCUSSION TITLE
How to ensure the same code formatting scheme?

### DISCUSSION THREAD

**Discussion by GOWxx** [sxyazi/yazi#1001] [1. Q&A] [answered]
Hello, I didn't seem to find any code formatting-related configuration or tools in the project source code. What method should I use to ensure consistent code formatting style? Thank you~

**Comment by sxyazi** [comment, marked as answer]
Yazi uses [`rustfmt`](https://github.com/rust-lang/rustfmt) nightly with this config https://github.com/sxyazi/yazi/blob/main/rustfmt.toml

**Reply by GOWxx** [reply]
Sorry to bother you again.

How is the toml file formatted in our project, using taplo?

If we use taplo, how do we upload a configuration file to the root directory? This way, we can have consistent formatting for toml file.

**Reply by sxyazi** [reply]
I use [`eslint-plugin-toml`](https://www.npmjs.com/package/eslint-plugin-toml) to format TOML files, since I find it more powerful than any existing formatter/linter. Here is my ESLint config https://github.com/sxyazi/dotfiles/blob/main/rules/eslint/plugins/toml.js

**Reply by GOWxx** [reply]
Thank you for sharing your `dotfiles`; they are truly impressive! The configurations you've set up are fascinating! 

However, I'm concerned that these personalized settings might pose a challenge for others who wish to contribute to the project. It could be difficult for new contributors to replicate your unique setup, which might hinder their ability to contribute effectively.

To address this, could you consider integrating linting and formatting tools and configurations directly into the project? This addition would help standardize the code style across different contributors and streamline the development process. It would particularly help in managing large-scale changes without the need for manual style adjustments.

Of course, other contributors can focus more on the functionality of the code itself and simply submit their contributions. I've noticed that sxyazi will help in unifying the code format, ensuring consistency throughout the project.

While the primary focus should remain on the functionality of the code, ensuring a consistent coding style could significantly enhance the contribution experience and accessibility of the project.

I look forward to seeing the project grow and wish you continued success!
😘😘😘
